### PR TITLE
LVJAVS-458 bug fixed

### PR DESCRIPTION
**Pull request by merkan3k**
LVJAVS-458 - Error messages don't appear when enter too short name in ‘First name’ and ‘Last name’ fields.

Validation occurs on the frontend - file edit-my-kid.jsp added scripts validation-edit-kid.js and validation-edit-kid - ua.js. We use appropriate scripte according to the localization;
on the backend - in the file ChildValidator.java

### CODE REVIEW

**Comment by vplytan-amplify** [comment]
please, write PR description in English

File: src/main/java/ua/softserveinc/tc/validator/ChildValidator.java [hunk 1]
```diff
@@ -38,6 +38,23 @@ public void validate(Object o, Errors errors) {
...
 
+
+/**
+ * check if the first and last names are corrent(if there is only 1 letter)
+ * Error messages appear. "First Name" and "Last Name" fields are not edited.
+ */
+        String fName = kidToValidate.getFirstName();
+        String lName = kidToValidate.getLastName();
+
+        if ((fName.length() < 2) || (lName.length() < 2)) {
```
Comment (vplytan-amplify) [inline src/main/java/ua/softserveinc/tc/validator/ChildValidator.java]: why do you need this condition ?

File: src/main/java/ua/softserveinc/tc/validator/ChildValidator.java [hunk 2]
```diff
@@ -38,6 +38,23 @@ public void validate(Object o, Errors errors) {
...
+
+            if(fName.length() < 2) {
```
Comment (vplytan-amplify) [inline src/main/java/ua/softserveinc/tc/validator/ChildValidator.java]: move magic number (2) to constant

File: src/main/webapp/resources/js/validation-edit-kid-ua.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * Created by Вова on 14.11.2016.
+ */
+$(document).ready(function() {
+    $(function() {
+        $.validator.addMethod("regexFirstName", function(value, element, regexpr) {
+            return regexpr.test(value);
+        }, "Все плохо");
+        $.validator.addMethod("regexLastName", function(value, element, regexpr) {
+            return regexpr.test(value);
+        }, "Все плохо");
```
Comment (vplytan-amplify) [inline src/main/webapp/resources/js/validation-edit-kid-ua.js]: what does it mean ?

File: src/main/webapp/resources/js/validation-edit-kid-ua.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+                    required: "Будь ласка введіть щонайменше 2 символи"
+
+                },
+                lastName: {
+                    required: "Будь ласка введіть щонайменше 2 символи"
+                }
+            }
+        });
+    });
+});
```
Comment (vplytan-amplify) [inline src/main/webapp/resources/js/validation-edit-kid-ua.js]: new line

File: src/main/webapp/resources/js/validation-edit-kid.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                },
+                lastName: {
+                    required: 'Please enter at least 2 characters'
+
+                }
+
+            }
+        });
+    });
+});
```
Comment (vplytan-amplify) [inline src/main/webapp/resources/js/validation-edit-kid.js]: new line

File: src/main/java/ua/softserveinc/tc/validator/ChildValidator.java (hunk 2 above)
Comment (merkan3k) [inline src/main/java/ua/softserveinc/tc/validator/ChildValidator.java]: done

File: src/main/webapp/resources/js/validation-edit-kid-ua.js (hunk 3 above)
Comment (merkan3k) [inline src/main/webapp/resources/js/validation-edit-kid-ua.js]: done

File: src/main/webapp/resources/js/validation-edit-kid-ua.js (hunk 4 above)
Comment (merkan3k) [inline src/main/webapp/resources/js/validation-edit-kid-ua.js]: done

File: src/main/webapp/resources/js/validation-edit-kid.js (hunk 5 above)
Comment (merkan3k) [inline src/main/webapp/resources/js/validation-edit-kid.js]: done

**Comment by hrybs** [comment]
You must change this pull request to merge it with master branch but not with dev branch.

File: src/main/java/ua/softserveinc/tc/validator/ChildValidator.java [hunk 6]
```diff
@@ -46,14 +49,12 @@ public void validate(Object o, Errors errors) {
         String fName = kidToValidate.getFirstName();
         String lName = kidToValidate.getLastName();
 
-        if ((fName.length() < 2) || (lName.length() < 2)) {
+        if(fName.length() < MIN_NAME_CHARACTER) {
+            errors.rejectValue(ValidationConstants.FIRST_NAME, ValidationConstants.NAME_NOT_EDITTED);
+        } else if(lName.length() < MIN_NAME_CHARACTER) {
+            errors.rejectValue(ValidationConstants.LAST_NAME, ValidationConstants.NAME_NOT_EDITTED);
+        }
```
Comment (VitalijF) [inline src/main/java/ua/softserveinc/tc/validator/ChildValidator.java]: Validation will happen only in one field. When  (fName.length < MIN_NAME_CHARACTER) gives true, else statement never happened even if lName.length < MIN_NAME_CHARACTER)
 I think that  you should write two independent if without else.
Example:
 if(fName.length() < MIN_NAME_CHARACTER){
// code
}
 if(lName.length() < MIN_NAME_CHARACTER){
// code
}

**Comment by hrybs** [comment]
There is something wrong, because I see my commits in your pull request. Either I am wrong or you must fix it.

File: src/main/java/ua/softserveinc/tc/validator/ChildValidator.java (hunk 6 above)
Comment (merkan3k) [inline src/main/java/ua/softserveinc/tc/validator/ChildValidator.java]: done

Comment (merkan3k) [inline src/main/java/ua/softserveinc/tc/validator/ChildValidator.java]: done

File: src/main/webapp/resources/js/validation-edit-kid-ua.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+/**
+ * Created by Вова on 14.11.2016.
+ */
+$(document).ready(function() {
+    $(function() {
+        $.validator.addMethod("regexFirstName", function(value, element, regexpr) {
+            return regexpr.test(value);
+        }, "Ви повинні ввести як мінімум два символи");
```
Comment (VitalijF) [inline src/main/webapp/resources/js/validation-edit-kid-ua.js]: spring:message doesn't work in JS files

**Pull request merged**